src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    public static final double GROW_MULTIPLIER = 1.5;
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size + 1 > elements.length) {
+            elements = grow();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index + " is out of bounds");
+        }
+        if (size >= elements.length) {
+            elements = grow();
+        }
+        if (size == index) {
+            elements[index] = value;
+            size++;
+            return;
+        }
+        Object[] temp = new Object[elements.length];
+        for (int i = 0; i < index; i++) {
+            temp[i] = elements[i];
+        }
+        temp[index] = value;
+        for (int i = index + 1; i < size + 1; i++) {
+            temp[i] = elements[i - 1];
+        }
```

Comment:
https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html#use-systemarraycopy-to-move-your-array-elements